core/src/main/java/org/ldaptive/url/RegexUrlParser.java

```diff
@@ -0,0 +1,118 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.ldaptive.url;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.ldaptive.LdapUtils;
+import org.ldaptive.SearchScope;
+import org.ldaptive.dn.DefaultDnParser;
+import org.ldaptive.dn.Dn;
+import org.ldaptive.dn.DnParser;
+import org.ldaptive.filter.Filter;
+import org.ldaptive.filter.FilterFunction;
+import org.ldaptive.filter.RegexFilterFunction;
+
+/**
+ * Parses an LDAP URL string using a regular expression.
+ *
+ * @author  Middleware Services
+ */
+public final class RegexUrlParser implements UrlParser
```

Comment:
Strongly recommend you provide exactly one implementation of `UrlParser` with the library. It's my understanding that `DefaultUrlParser` is more performant; perhaps that should be the implementation you ship.

Reply:
It becomes the default implementation with this patch. I'd prefer not to throw away the regex parser just yet, probably in a future release.

Reply:
Suggest you leverage `@Deprecated` if you thinking of future removal.

Reply:
Made a note for future deprecation.